cmake/HalideGeneratorHelpers.cmake

```diff
@@ -57,6 +59,18 @@ function(add_halide_generator TARGET)
         add_executable(${gen} ALIAS ${TARGET})
         target_link_libraries(${TARGET} PRIVATE Halide::Generator ${ARG_LINK_LIBRARIES})
 
+        if (NOT ARG_NO_DEFAULT_FLAGS AND NOT Halide_NO_DEFAULT_FLAGS)
+            # For crosscompiling builds, the Halide headers will be included using -isystem,
+            # which will cause all warnings to be ignored. This is not helpful, since
+            # we *want* deprecation warnings to be propagated. So we must set
+            # NO_SYSTEM_FROM_IMPORTED in order for it to be seen.
+            set_target_properties(${TARGET} PROPERTIES NO_SYSTEM_FROM_IMPORTED YES)
```

Comment:
It seems that Clang has a special flag for this... https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-in-system-headers

The other approach is to try to make the deprecation warning fire on the class that is directly instantiated by the user, so that the warning appears to originate from user code.